python/dgl/nn/pytorch/explain/subgraphxexplainer.py

```diff
@@ -0,0 +1,552 @@
+        return the biggest one.
+
+        Parameters
+        ----------
+        graph: DGLGraph
+            A homogeneous graph.
+
+        Returns
+        -------
+        Function
+            Returns the largest weakly connected components in graph.
+        """
+
+        # Ensure graph is homogenous.
+        new_graph = dgl.to_homogeneous(graph)
+        # Turn graph into a bidirected graph
+        new_graph = dgl.to_bidirected(new_graph)
+        # Convert to a networkx graph object
+        nx_graph = dgl.to_networkx(new_graph).to_undirected()
+        # Find and sort graph components by size from largest to smallest and take the biggest one.
+        biggest_comp = list(
+            sorted(nx.connected_components(nx_graph), key=len, reverse=True)
+        )[0]
+        # Convert back to DGLGraph object.
+        return list(biggest_comp)
+
+    def prune_graph(self, graph, strategy):
+        r"""Find the graph based on the chosen strategy. Once prunes, return the
+        list of subgraphs, list of subgraph nodes and list of pruned nodes.
+
+        Parameters
+        ----------
+        graph: DGLGraph
+            A homogeneous graph.
+        strategy: str
+            The strategy based on which the pruning will happen, "High2low" or "Low2high".
+
+        Returns
+        -------
+        tuple(list, list, list)
+            Return a tuple consists of list of subgraphs based on pruned action,
+            list of subgraphs nodes, and list of pruned nodes.
+
+        """
+        subgraphs = []
+        subgraphs_nodes_mapping = []
+
+        rev = False
+        if strategy == "High2low":
+            rev = True
+
+        out_degrees = graph.out_degrees()
+        in_degrees = graph.in_degrees()
+        nodes = graph.nodes()
+        node_degree = []
+
+        for node, in_degree, out_degree in zip(nodes, in_degrees, out_degrees):
+            node_degree.append((node, in_degree + out_degree))
+
+        node_degree = sorted(node_degree, key=lambda x: x[1], reverse=rev)
+
+        if len(node_degree) < self.num_child_expand:
+            pruned_nodes = nodes
+        else:
+            pruned_nodes = [
+                node_degree[i][0] for i in range(self.num_child_expand)
+            ]
+
+        for prune_node in pruned_nodes:
+            new_subgraph_nodes = np.array(
+                [node for node in nodes if node != prune_node]
+            )
+            new_subgraph = dgl.node_subgraph(graph, new_subgraph_nodes)
+            biggest_comp = self.biggest_weak_component(new_subgraph)
+            new_subgraph_big_comp = dgl.node_subgraph(graph, biggest_comp)
+
+            subgraphs_nodes_mapping.append(new_subgraph_nodes[biggest_comp])
+            subgraphs.append(new_subgraph_big_comp)
+
+        return subgraphs, subgraphs_nodes_mapping, pruned_nodes
+
+    def explain_graph(
+        self, graph, features, num_gnn_layers=4, mc_sampling_steps=100, **kwargs
```

Comment:
If `num_gnn_layers` needs to be the same as that used by the GNN model, then we should not provide a default value and this should be set in `__init__`.